.gitlab/ci/build_quartz.yml

```diff
@@ -4,62 +4,79 @@
   tags:
     - shell
     - quartz
-  except:
-    refs:
-      - /_qnone/
-    variables:
-      - $SERAC_CI_QUARTZ == "OFF"
+  rules:
+    - if: '$CI_COMMIT_BRANCH =~ /_qnone/ || $ON_QUARTZ == "OFF"' #run except if ...
+      when: never
+    - if: '$CI_JOB_NAME =~ /release_resources/'
+      when: always
+    - when: on_success
 
 ####
 # In pre-build phase, allocate a node for builds
 allocate_resources_build_quartz:
   variables:
     GIT_STRATEGY: none
-  extends: .on_quartz
+  extends: [.on_quartz, .pr_workflow]
   stage: allocate_resources
   script:
-    - salloc -N 1 -c 36 -p pdebug -t 30 --no-shell --job-name=${PROJECT_ALLOC_NAME}
+    - salloc ${ACCOUNT} -N 1 -c 36 -t 30 --no-shell --job-name=${PROJECT_ALLOC_NAME}
 
 ####
 # In post-build phase, deallocate resources
 # Note : make sure this is run even on build phase failure
 release_resources_build_quartz:
   variables:
     GIT_STRATEGY: none
-  extends: .on_quartz
+  extends: [.on_quartz, .pr_workflow]
   stage: release_resources
   script:
     - export JOBID=$(squeue -h --name=${PROJECT_ALLOC_NAME} --format=%A)
-    - ([[ -n "${JOBID}" ]] && scancel ${JOBID})
-  when: always
+    - if [[ -n "${JOBID}" ]]; then scancel ${JOBID}; fi
 
 ####
 # Template
-.build_quartz:
-  extends: .on_quartz
+.pr_build_on_quartz:
   stage: build
-  script:
-    - JOBID=$(squeue -h --name=${PROJECT_ALLOC_NAME} --format=%A)
-    - JOBID=$(if [[ -n "${JOBID}" ]]; then echo "--jobid=${JOBID}"; fi)
-    - EXEC_PREFIX="srun ${JOBID} -t 20 -N 1 -n 1 -c 4"
-    - ${EXEC_PREFIX} scripts/gitlab/build_and_test.sh
+  extends: [.srun_build_script, .on_quartz, .pr_workflow]
+
+.master_build_with_deps_on_quartz:
```

Comment:
This will definitely be nice to have (at least occasionally)!

Reply:
Currently, it’s setup to do this each time master is changed.
However, master is not mirrored... Most likely, someone with write access to master does not have 2 Factors Authentication (or is not in LLNL org?).